driver-core/src/test/functional/com/mongodb/operation/DropIndexOperationSpecification.groovy

```diff
@@ -93,7 +95,15 @@ class DropIndexOperationSpecification extends OperationFunctionalSpecification {
         indexes[0].name == '_id_'
 
         where:
-        async << [true, false]
+        [keys, async] << [
+                [new BsonDocument('theField', new BsonInt32(1)),
+                 new BsonDocument('theField', new BsonInt32(1)).append('theSecondField', new BsonInt32(-1)),
+                 new BsonDocument('theField', new BsonString('2d')),
+                 new BsonDocument('theField', new BsonString('hashed')),
+//                 new BsonDocument('theField', new BsonString('text'))  // Server does not allow this
```

Comment:
See latest comment on the Jira issue.  It's one place where this could break existing applications, since you can delete a text index by its generated name, assuming that no non-default name was supplied at creation time.